src/renderer/dx/DxFontInfo.cpp

```diff
@@ -202,6 +202,63 @@ void DxFontInfo::SetFromEngine(const std::wstring_view familyName,
     return face;
 }
 
+// Routine Description:
+// - Locates a suitable font face from the given information using more advanced match filtering on font sets only
+//   possible with Win10 and higher versions of DirectWrite
+// Arguments:
+// - fontCollection - The font collection to dig through
+// - localeName - The locale to search for appropriate fonts.
+// Return Value:
+// - Smart pointer holding interface reference for queryable font data.
+// - The pointer can be empty/null if there was a failure or we're below Win10 and need to fallback
+[[nodiscard]] Microsoft::WRL::ComPtr<IDWriteFontFace1> DxFontInfo::_TryWin10FontMatchingLogic(gsl::not_null<IDWriteFontCollection*> fontCollection,
+                                                                                              std::wstring& localeName)
+{
+    Microsoft::WRL::ComPtr<IDWriteFontFace1> fontFace;
+
+    try
+    {
+        Microsoft::WRL::ComPtr<IDWriteFontCollection1> collection1;
+        THROW_IF_FAILED(fontCollection->QueryInterface(collection1.GetAddressOf()));
+
+        Microsoft::WRL::ComPtr<IDWriteFontSet> set;
+        THROW_IF_FAILED(collection1->GetFontSet(set.GetAddressOf()));
+
+        Microsoft::WRL::ComPtr<IDWriteFontSet1> set1;
+        THROW_IF_FAILED(set.As(&set1));
+
+        const std::array<DWRITE_FONT_PROPERTY, 3> fontProperties{
+            DWRITE_FONT_PROPERTY{ DWRITE_FONT_PROPERTY_ID_TYPOGRAPHIC_FAMILY_NAME, _familyName.data(), nullptr },
+            DWRITE_FONT_PROPERTY{ DWRITE_FONT_PROPERTY_ID_WIN32_FAMILY_NAME, _familyName.data(), nullptr },

```

Comment:
Sorry, I left out

```
DWRITE_FONT_PROPERTY_ID_TYPOGRAPHIC_FAMILY_NAME
DWRITE_FONT_PROPERTY_ID_WEIGHT_STRETCH_STYLE_FAMILY_NAME <<<
DWRITE_FONT_PROPERTY_ID_WIN32_FAMILY_NAME
DWRITE_FONT_PROPERTY_ID_FULL_NAME
```

Reply:
I noticed also that there is a `DWRITE_FONT_PROPERTY_ID_WEIGHT` but it wants... a string. And I have either the enum value for the old style or a variable axes value there. How do I use those ones?

Reply:
> Sorry, I left out...
> 
> ```
> DWRITE_FONT_PROPERTY_ID_TYPOGRAPHIC_FAMILY_NAME
> DWRITE_FONT_PROPERTY_ID_WEIGHT_STRETCH_STYLE_FAMILY_NAME <<<----
> DWRITE_FONT_PROPERTY_ID_WIN32_FAMILY_NAME
> DWRITE_FONT_PROPERTY_ID_FULL_NAME
> ```
> 
> ...which is usually the same as the typographic name anyway, but there are some rare cases where it's not.

This does not help find "Cascadia Mono Heavy" :(

Reply:
Yeah, please avoid "DWRITE_FONT_PROPERTY_ID_WEIGHT" and pass `DWRITE_FONT_AXIS_TAG_WEIGHT` to GetMatchingFonts. Regarding "Cascadia Mono Heavy", I'd have to look at the font before I can say more.

Reply:
Cascadia Mono doesn't have a Heavy weight because it's STAT table/fvar are missing such an entry. Please connect with the font designer Aaron Bell to fix the font bug.

![image](https://user-images.githubusercontent.com/1809166/127117724-d4e7b6e7-882e-4b31-9a07-bfa4889b5b9a.png)

Reply:
I believe that we didn't name a black instance by design. Cascadia's bold is _pretty bold_, after all.